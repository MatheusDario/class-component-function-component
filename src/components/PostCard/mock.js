export default {
  title: 'Title 1',
  cover: 'images/cover.png',
  body: `Lorem ipsum dolor sit amet consectetur adipisicing elit. Modi, fugit
  alias minus accusamus in incidunt facilis rerum delectus assumenda illo!
  Quae quia soluta eaque itaque temporibus libero eligendi odio architecto.`,
};
