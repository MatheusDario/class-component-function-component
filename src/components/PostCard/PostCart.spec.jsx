import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { PostCard } from '.';

import { props } from './mock';

describe('<PostCard />', () => {
  it('should correct component postcard', () => {
    renderTheme(<PostCard {...props} />);
    expect(screen.getByRole('img')).toBeInTheDocument();
    expect(screen.getByRole('heading')).toBeInTheDocument();
  });

  it('shuld match snapshot', () => {
    const { container } = renderTheme(<PostCard {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
