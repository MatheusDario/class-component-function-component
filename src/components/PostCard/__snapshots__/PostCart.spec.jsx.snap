// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> shuld match snapshot 1`] = `
.c0 {
  background-color: #cecece;
  box-shadow: 0px 0px 10px rgba(255, 255, 255, .4 );
  border-radius: 10px;
  margin: 1rem 0;
  transition: all 300ms ease-in-out;
  max-width: 50vh;
}

.c0:hover {
  transform: scale(1.05);
}

.c1 {
  padding: 1.5rem;
}

.c3 {
  font-size: 2.3rem;
}

.c4 {
  text-align: justify;
  font-size: 1.3rem;
}

.c2 {
  width: 100%;
}

<section
  class="c0"
>
  <div
    class="c1"
  >
    <img
      class="c2"
    />
    <h1
      class="c3"
    />
    <p
      class="c4"
    />
  </div>
</section>
`;
